packages/api/src/router/role.ts

```diff
@@ -2,23 +2,79 @@ import type { TRPCRouterRecord } from "@trpc/server";
 import { z } from "zod";
 
 import type { ReviewType } from "@cooper/db/schema";
-import { desc, eq } from "@cooper/db";
+import { asc, desc, eq } from "@cooper/db";
 import { CreateRoleSchema, Review, Role } from "@cooper/db/schema";
 
-import { protectedProcedure, publicProcedure } from "../trpc";
+import {
+  protectedProcedure,
+  sortableProcedure,
+  publicProcedure,
+} from "../trpc";
+
+const ordering = {
+  default: desc(Role.id),
+  newest: desc(Role.createdAt),
+  oldest: asc(Role.createdAt),
+};
 
 export const roleRouter = {
-  list: publicProcedure.query(({ ctx }) => {
+  list: sortableProcedure.query(async ({ ctx }) => {
```

Comment:
Ok... I know this is so unlike us but what are our thoughts on adding like, 2 tests to see if this works. Just because I don't want to wait to implement the frontend for this, otherwise this is good